### PR TITLE
Several Updates (AA seq import/export; Generative Model Submission; BULK Test improvements; Common Client improvement)

### PR DESCRIPTION
**Pull request by AndresPerezTesela**
Some updates have been done in this PR listed by corresponding Teselagen Module below:

DESIGN:
- Support import/export of amino acid sequences.

TEST:
- Supports bulk GET/DELETE of Assay subjects.

DISCOVER:
- New Generative Model submission function implemented.
- Multi Objective Optimization function implemented.

COMMON:
- All TG Clients can be instantiated with a common client instance for common endpoint sharing.

### CODE REVIEW

File: lib/teselagen/api/design_client.py [hunk 1]
```diff
@@ -432,3 +454,198 @@ def rbs_calculator_submit_job(self, algorithm: str, params: dict)->dict:
...
+    ## Amino acid sequence Methods.
+    @requires_login
+    def import_aa_sequences(self, aa_sequences: Union[pd.DataFrame, List[List[str]], List[Tuple[str, str]], List[Dict[str,str]]], tags: Optional[List[Dict[str, int]]] = None):
+        '''
+        This function imports one or many amino acid sequences by means of TeselaGen's DESIGN API.
+
+        Args:
+            aa_sequences(Union[pd.DataFrame, List[Dict[str,str]], List[Tuple[str, str]]): Amino acid sequences data. The data can come in three different ways:
+                - as a pandas dataframe with 2 columns. Where the first column contains the sequence names and the second column contains the amino acid sequence string.
+                - as a list of python dictionaries, where each dictionary is of the form `{"AA_NAME": SEQUENCE_NAME, "AA_SEQUENCE": SEQUENCE_STRING}`.
```
Comment (diegovalenzuelaiturra) [inline lib/teselagen/api/design_client.py:475]: Perhaps rename `"AA_NAME"` by `"AA_SEQUENCE_NAME"`

File: lib/teselagen/api/discover_client.py [hunk 2]
```diff
@@ -505,69 +526,84 @@ def design_crispr_grnas(self,
...
+        if (aa_sequences is not None):
+            if isinstance(aa_sequences, list) or isinstance(aa_sequences, np.ndarray):
+                if all(isinstance(x, str) for x in aa_sequences):
+                    kwargs['data_input'] = list(map(lambda x: {'sequence': x}, aa_sequences))
+                else:
+                    raise ValueError("All amino acid sequences must be of type string.")
+        elif (aa_sequence_ids is not None):
+            if isinstance(aa_sequence_ids, list) or isinstance(aa_sequence_ids, np.ndarray):
+                if all(isinstance(x, int) for x in aa_sequence_ids):
+                    pass
```
Comment (diegovalenzuelaiturra) [inline lib/teselagen/api/discover_client.py]: Is it ok to `pass` or should we raise a `NotImplementedError` or something ?

File: lib/teselagen/api/test_client.py [hunk 3]
```diff
@@ -25,53 +25,59 @@ def __init__(self,
                  api_token_name: str = DEFAULT_API_TOKEN_NAME,
                  host_url: str = DEFAULT_HOST_URL):
         module_name: str = "test"
-        super(TESTClient, self).__init__(module_name=module_name,
-                                         host_url=host_url,
-                                         api_token_name=api_token_name)
+        if (tg_client is not None):
```
Comment (diegovalenzuelaiturra) [inline lib/teselagen/api/test_client.py]: I think `tg_client` is not defined

**Review by diegovalenzuelaiturra — COMMENTED** [review]
Looks good.

As a comment, I think most abbreviations could be avoided when writing code, and in general, they should. I'd personally try to write code using variable names as unambiguously as possible. I'd suggest we replace `aa` by `amino_acid`, `aa_name` by `amino_acid_sequence_name`, etc)

**Review by diegovalenzuelaiturra — APPROVED** [review]

**Review by juanneilson — APPROVED** [review]
Nice!

**Pull request merged**